packages/ui/src/components/label-marker.tsx

```diff
@@ -25,23 +29,71 @@ export const LabelMarkerColor = {
+  [ColorsEnum.YELLOW]: 'text-label-foreground-yellow hover:opacity-60',
+  [ColorsEnum.PURPLE]: 'text-label-foreground-purple hover:opacity-60',
+  [ColorsEnum.PINK]: 'text-label-foreground-pink hover:opacity-60',
+  [ColorsEnum.VIOLET]: 'text-label-foreground-violet hover:opacity-60',
+  [ColorsEnum.INDIGO]: 'text-label-foreground-indigo hover:opacity-60',
+  [ColorsEnum.CYAN]: 'text-label-foreground-cyan hover:opacity-60',
+  [ColorsEnum.ORANGE]: 'text-label-foreground-orange hover:opacity-60',
+  [ColorsEnum.BROWN]: 'text-label-foreground-brown hover:opacity-60',
+  [ColorsEnum.MINT]: 'text-label-foreground-mint hover:opacity-60',
+  [ColorsEnum.LIME]: 'text-label-foreground-lime hover:opacity-60'
+}
+
+export const LabelMarker: FC<LabelMarkerProps> = ({
+  type = LabelType.STATIC,
+  label,
+  value,
+  color,
+  counter,
+  onDelete,
+  className
+}) => {
+  const isDynamic = type === LabelType.DYNAMIC
+  const isWithExtraContent = !!value || !!counter
+  const isWithDeleteButton = !!onDelete
 
-export const LabelMarker: FC<LabelMarkerProps> = ({ label, value, color, inTable = false }) => {
   return (
-    <span
-      className={cn(
-        `inline-flex text-12 leading-5 h-5 py-px px-2 items-center font-medium rounded overflow-hidden max-w-full ${LabelMarkerColor[color]}`,
-        { 'pr-px': !!value }
-      )}
-    >
-      <span className={cn('truncate', { 'flex-1': inTable }, { 'pr-1.5': !!value })}>{label}</span>
-      {!!value && (
-        <span
-          className={`flex h-full items-center overflow-hidden rounded-r bg-label-background-black pl-1.5 pr-[7px] ${inTable && 'max-w-[50%]'}`}
+    <div className={cn('flex max-w-full items-center gap-2 opa relative', className)}>
+      <span
+        className={cn(
+          `grid text-12 leading-4 h-5 py-px px-2 items-center font-medium rounded overflow-hidden w-fit ${LabelMarkerColor[color]}`,
+          { 'pr-px grid-cols-[auto,auto]': isWithExtraContent },
+          { 'pr-6': !isWithExtraContent && isWithDeleteButton }
+        )}
+      >
+        <span className={cn('truncate', { 'pr-1.5': isWithExtraContent })}>{label}</span>
+
+        {isWithExtraContent && (
+          <span
+            className={cn(
+              `bg-label-background-black flex h-full items-center overflow-hidden rounded-r-[3px] pl-1.5 pr-[7px]`,
+              { 'pr-6': isWithExtraContent && isWithDeleteButton }
+            )}
+          >
+            {!!value && <span className="max-w-full truncate">{value}</span>}
+
+            {!!counter && counter}
+          </span>
+        )}
+      </span>
+
+      {isWithDeleteButton && (
+        <Button
+          className={`absolute right-1 top-0.5 max-h-4 min-h-4 min-w-4 max-w-4 px-0 ${deleteButtonColor[color]}`}
+          variant="custom"
+          onClick={onDelete}
         >
-          <span className="max-w-full truncate">{value}</span>
-        </span>
+          <Icon name="cross" size={10} />
+        </Button>
```

Comment:
Can we add something here for screen readers to pick up? Either an `aria-label` on the `Button`, or a string as a sibling of the `Icon` with a class of `sr-only`?